src/modules/mavlink/mavlink_high_latency2.cpp

```diff
@@ -151,10 +158,24 @@ bool MavlinkStreamHighLatency2::send(const hrt_abstime t)
 				_airspeed_sp.get_scaled(msg.airspeed_sp, 5.0f);
 			}
 
-			if (_battery.valid()) {
-				_battery.get_scaled(msg.battery, 100.0f);
+			int lowest = -1;
+
+			for (int i = 0; i < BOARD_NUMBER_BRICKS; i++) {
+				if (_battery_connected[i] && (lowest == -1 ||
+							      (_battery[i].valid() && _battery[i].get_scaled(100.0f) <= _battery[lowest].get_scaled(100.0f)))) {
```

Comment:
This is hard to read. If you can extract an intermediate variable and give it a name it would help.